kotlinlib/src/mill/kotlinlib/KotlinModule.scala

```diff
@@ -102,19 +102,53 @@ trait KotlinModule extends JavaModule { outer =>
     )().toSeq ++ kotlinWorkerClasspath()
   }
 
+  /**
+   * Flag to use the embeddable kotlin compiler.
+   * This can be necessary to avoid classpath conflicts or ensure
+   * compatibility to the used set of plugins.
+   *
+   * See also https://discuss.kotlinlang.org/t/kotlin-compiler-embeddable-vs-kotlin-compiler/3196
+   */
+  def kotlinCompilerEmbeddable: Task[Boolean] = Task { false }
+
+  /**
+   * The kotlin-compiler dependency.
+   *
+   * It uses the embeddable version, if [[kotlinCompilerEmbeddable]] is `true`.
+   */
+  def kotlinCompilerDep: T[Dep] = Task {
```

Comment:
We probably should make this return a `Seq[Dep]`, in case, multiple deps or more control is needed.